### ISSUE TITLE
[rule] Option for quotes rule to allow JSON-like object literals

### ISSUE THREAD

**Issue by epoberezkin** [eslint/eslint#6906]
**What version of ESLint are you using?**

3.3.0

**What rule do you want to change?**

Rule: quotes
Possible change:
Option `allowJSONLiterals: true` in `[ 2, "single" ]` rule

**What code should be flagged as incorrect with this change?**

The idea is to allow double quotes in object literals if they are used consistently to be a valid JSON.

With `[ 2, "single", "allowJSONLiterals": true ]`

Correct :

``` javascript
var a = { "a": "b" };
var c = 'd';
```

Also correct:

``` javascript
var a = { a: 'b' };
var c = 'd';
```

Incorrect:

``` javascript
var a = { a: "b" };
var c = 'd';
```

**What happens when the rule is applied to this code now?**

With `[ 2, "single" ]`

Incorrect:

``` javascript
var a = { "a": "b" };
var c = 'd';
```

Correct:

``` javascript
var a = { a: 'b' };
var c = 'd';
```

**Comment by platinumazure** [comment]
Why do you need this? You can always use `JSON.stringify` to generate the correct string with the correct quotes when you actually need to serialize into JSON strings.

Could you please explain your use case more thoroughly? Thanks!

**Comment by epoberezkin** [comment]
It is intended as stylistic rule to allow JSON-like objects in code like here: https://github.com/epoberezkin/ajv/blob/master/spec/async.spec.js#L11

It also simplifies refactoring when JSON file has to be included in code or when object has to be separated in JSON file.

**Comment by gajus** [comment]
In my opinion, this introduces code style inconsistency. I do not see a good reason to allow JSON-like object definitions in JavaScript code purely for stylistic purpose.

I can see how this would easily enable developers to start "preferring" writing objects in JSON style for no good reason.

> Why do you need this? You can always use JSON.stringify to generate the correct string with the correct quotes when you actually need to serialize into JSON strings.

This is irrelevant. Functionally, both object declarations are equal. I am guessing that the sole purpose of this rule is to enable an easy copy-paste routine from JavaScript to JSON and vice-versa. As such, it is a very esoteric requirement.

I therefore recommend:

* Store JSON object definitions in a separate file and import them as needed. As a plus side, this will enable use of JSON linter and JSON formatter to keep your JSON structures tidy; or
* Use `/* eslint-disable */` in combination with `/* eslint-enable */` to disable and enable specific style requirements for blocks of code.

**Comment by kaicataldo** [comment]
I'm closing this issue because it looks like consensus couldn't be reached. While we wish we could accommodate all requests, we have limited resources and do need to prioritize. We've found that issues failing to reach consensus after 21 days tend to never reach consensus, and as a team have decided to close such issues. This doesn't mean the idea isn't interesting or valuable, only that it's not something the team can commit to at the moment.
